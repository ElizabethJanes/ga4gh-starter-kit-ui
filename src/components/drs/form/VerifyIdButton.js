import React from 'react';
import { 
    Typography,
    Button,
    IconButton,
    Tooltip
} from '@material-ui/core';
import CheckCircle from '@material-ui/icons/CheckCircle';
import Cancel from '@material-ui/icons/Cancel';

const VerifyIdButton = props => {

    const validateRelative = () => {

    }

    const callAndValidate = () => props.retrieveDrsObject(props.relative.id, validateRelative);

    return (
        <div>
            <Button
                color='primary'
                disabled={props.readOnly}
                variant='contained'
                size='small'
                onClick={callAndValidate}
            >
                <Tooltip title='Submit this ID for verification.'>
                    <Typography variant='button'>Verify ID</Typography>
                </Tooltip>
            </Button>

            {props.relative.isValid === true
                ?
                    <IconButton color='primary'>
                        <Tooltip title='This is a valid ID.'>
                            <CheckCircle/>
                        </Tooltip>
                    </IconButton>
                : null
            }

            {props.relative.isValid === false
                ?
                    <IconButton color='secondary'>
                        <Tooltip title='This is an invalid ID. Please enter a valid ID before proceeding.'>
                            <Cancel/>
                        </Tooltip>
                    </IconButton>
                : null
            }
        </div>
    )
}

export default VerifyIdButton;
