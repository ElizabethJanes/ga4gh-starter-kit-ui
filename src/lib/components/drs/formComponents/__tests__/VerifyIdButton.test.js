import React from 'react';
import renderer from 'react-test-renderer';
import VerifyIdButton from '../VerifyIdButton';

test('<VerifyIdButton/> should handle valid relatives', () => {
    const verifyIdButton = renderer
    .create(<VerifyIdButton relative={{isValid: true}} />)
    .toJSON();
    expect(verifyIdButton).toMatchSnapshot();
})

test('<VerifyIdButton/> should handle invalid relatives', () => {
    const verifyIdButton = renderer
    .create(<VerifyIdButton relative={{isValid: false}} />)
    .toJSON();
    expect(verifyIdButton).toMatchSnapshot();
})

test('<VerifyIdButton/> should handle undefined relatives', () => {
    const verifyIdButton = renderer
    .create(<VerifyIdButton relative={{isValid: undefined}} />)
    .toJSON();
    expect(verifyIdButton).toMatchSnapshot();
})