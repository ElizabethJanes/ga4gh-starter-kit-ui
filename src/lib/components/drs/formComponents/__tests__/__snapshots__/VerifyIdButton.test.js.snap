// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifyIdButton/> should handle invalid relatives 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        aria-describedby={null}
        className="MuiTypography-root MuiTypography-button"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Submit this ID for verification."
      >
        Verify ID
      </span>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-describedby={null}
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="This is an invalid ID. Please enter a valid ID before proceeding."
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`<VerifyIdButton/> should handle undefined relatives 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        aria-describedby={null}
        className="MuiTypography-root MuiTypography-button"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Submit this ID for verification."
      >
        Verify ID
      </span>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`<VerifyIdButton/> should handle valid relatives 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        aria-describedby={null}
        className="MuiTypography-root MuiTypography-button"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Submit this ID for verification."
      >
        Verify ID
      </span>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-describedby={null}
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="This is a valid ID."
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
